WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardView.swift

```diff
@@ -21,7 +21,11 @@ final class BlazeCampaignDashboardViewHostingController: SelfSizingHostingContro
         }
 
         rootView.createCampaignTapped = { [weak self] in
-            self?.navigateToCampaignCreation(source: .myStoreSection)
+            if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.blazei3NativeCampaignCreation) {
+                // No need to do anything since we're using .sheet to navigate.
+            } else {
+                self?.navigateToCampaignCreation(source: .myStoreSection)
+            }
```

Comment:
The design hints that we'll need to push the campaign creation form to the same navigation stack after selecting a product from the list, so I think using sheet is not the best way to handle this. Due to the complication, we should use UIKit for the navigation. I'd suggest removing the sheet and handle the navigation here:

```suggestion
            guard let self else { return }
            if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.blazei3NativeCampaignCreation) {
                if self.viewModel.shouldShowProductSelectorView {
                    // navigate to the product selector view using hosting controller
                } else {
                    // navigate to the campaign creation form with the single existing product found in the store.
                }
            } else {
                self.navigateToCampaignCreation(source: .myStoreSection)
            }
```